proto/prysm/storage/beacon_storage.proto

```diff
@@ -0,0 +1,29 @@
+// Copyright 2021 Prysmatic Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+syntax = "proto3";
+
+package ethereum.eth.storage;
+
+import "proto/eth/v2/version.proto";
+
+
+// BeaconStateForStorage is used only inside db/kv package and it there for the sole purpose of
+// storing the state objects efficiently. With normalizing the fields in the "state" bucket
+// it also stores some indexes and other metadata fields that cannot be stored in the
+// BeaconState object itself.
+message BeaconStateForStorage {
+  v2.Version version = 1001;        // Used to identify the state object
```

Comment:
This type makes me a little nervous. It is an enum without an "undefined" zero value. For context, in protobuf it is a best practice to define the 0 index of an enum as undefined so that it can be differentiated from a value that is accidentally empty. `v2` is also a package for API types, importing it into v1alpha1 feels a little out of place. I wonder if we should define a new enum for database storage so that we can follow the best practice for safety.

Reply:
ok for not having 0 as a valid enum.
but I disagree to have multiple versions types in multiple packages. 
like phase0 is 0 in API and 1 in storage package.

Reply:
They are distinct strongly typed values in separate packages, and they have different uses. One is used by the API and the other by DB unmarshaling code - different envelopes, different layers of the stack (in other words, you aren't ever going to send the storage format over the wire in the rpc package). So it is probably ok for them to be different.

Reply:
@nisdas what do you think. Should we have a separate version for storage?

Reply:
@jmozah I dont have any issues with that. It maybe better to not mix up API and DB versioning here. The 0 value is a valid point as there would be no way to distinguish it from a zero value otherwise. We only plan to use this in the DB package in any case.

Reply:
> They are distinct strongly typed values in separate packages, and they have different uses.

@kasey I think you have a point. I will use a different version for storage and not start that with 0.

> The 0 value is a valid point as there would be no way to distinguish it from a zero value otherwise.

@nisdas All our other versions do have a valid value for 0.  ex: Phase0. Not sure if this applies there too.

Reply:
> All our other versions do have a valid value for 0. ex: Phase0. Not sure if this applies there too.

One of those is determined via the spec apis, we don't have control over those unfortunately. The other isn't as tied to the specs I think since it is an internal prysm value.